### PR TITLE
Chore/Bump Rust toolchain to 1.70.0

### PR DESCRIPTION
**Pull request by bsbds**
Please briefly answer these questions:

* what problem are you trying to solve? (or if there's no problem, what's the motivation for this change?)

    Described in #336 

* what changes does this pull request make?

    Bump Rust toolchain to 1.70.0

* are there any non-obvious implications of these changes? (does it break compatibility with previous versions, etc)

### CODE REVIEW

File: benchmark/src/runner.rs [hunk 1]
```diff
@@ -67,30 +68,22 @@ impl Stats {
     }
 
     /// Summary of stats
+    #[allow(clippy::let_underscore_untyped)]
```
Comment (Phoenix500526) [inline benchmark/src/runner.rs:71]: In fact, we can remove this lint.
```suggestion
    /// Summary of stats
    #[allow(clippy::let_underscore_must_use)] // the writeln! will not fail thus always return an Ok
    pub fn summary(&self) -> String {
        let mut s = String::from("\nSummary:\n");
        _ = writeln!(s, "  Total:        {:.4} secs", self.total.as_secs_f64());
        _ = writeln!(s, "  Slowest:      {:.4} secs", self.slowest.as_secs_f64());
        _ = writeln!(s, "  Fastest:      {:.4} secs", self.fastest.as_secs_f64());
        _ = writeln!(s, "  Average:      {:.4} secs", self.avg.as_secs_f64());
        _ = writeln!(s, "  Requests/sec: {:.2}", self.qps);
        s
    }
```

File: benchmark/src/runner.rs [hunk 2]
```diff
@@ -67,30 +68,22 @@ impl Stats {
...
+        let _ = writeln!(s, "  Average:      {:.4} secs", self.avg.as_secs_f64());
+        let _ = writeln!(s, "  Requests/sec: {:.2}", self.qps);
         s
     }
 
     /// Calculate the histogram from the latencies.
     #[allow(clippy::indexing_slicing)]
+    #[allow(clippy::let_underscore_untyped)]
+    #[allow(clippy::let_underscore_must_use)] // the writeln! will not fail thus always return an Ok
     pub fn histogram(&self) -> String {
```
Comment (Phoenix500526) [inline benchmark/src/runner.rs:87]: Same as above

File: benchmark/src/runner.rs [hunk 3]
```diff
@@ -208,7 +202,7 @@ impl CommandRunner {
             let tx_clone = tx.clone();
             let handle = tokio::spawn(async move {
                 let mut key = vec![0u8; key_size];
-                let _ = c.wait().await;
+                let _: tokio::sync::BarrierWaitResult = c.wait().await;
```
Comment (Phoenix500526) [inline benchmark/src/runner.rs:205]: We can modify this line of code as `_ = c.wait().await`. It's more concise.

File: benchmark/src/runner.rs [hunk 4]
```diff
@@ -268,7 +262,7 @@ impl CommandRunner {
             });
         }
         let mut stats = Stats::new();
-        let _ = barrier.wait().await;
+        let _: tokio::sync::BarrierWaitResult = barrier.wait().await;
```
Comment (Phoenix500526) [inline benchmark/src/runner.rs:265]: Same as above

File: xline/src/server/command.rs [hunk 5]
```diff
@@ -89,7 +89,7 @@ impl KeyRange {
     #[inline]
     pub fn new_one_key(key: impl Into<Vec<u8>>) -> Self {
         let key_vec = key.into();
-        assert!(key_vec.as_slice() != UNBOUNDED);
+        assert!(key_vec.as_slice() != UNBOUNDED, "key cannot be UNBOUNDED");
```
Comment (Phoenix500526) [inline xline/src/server/command.rs:92]: This diagnostic message doesn't provide any useful information.  When the assertion fails, we are more concerned about what type this key is and why it is of that type.

File: xline/src/storage/index.rs [hunk 6]
```diff
@@ -222,10 +222,13 @@ impl IndexOperate for Index {
...
-            assert!(inner
-                .unavailable_cache
-                .insert(revision, keys.clone())
-                .is_none());
+            assert!(
+                inner
+                    .unavailable_cache
+                    .insert(revision, keys.clone())
+                    .is_none(),
+                "revision {revision} already in unavailable cache",
```
Comment (Phoenix500526) [inline xline/src/storage/index.rs:230]: "revision {revision} is already in the unavailable cache",

File: benchmark/src/runner.rs (hunk 1 above)
Comment (bsbds) [reply benchmark/src/runner.rs]: done

File: benchmark/src/runner.rs (hunk 4 above)
Comment (bsbds) [reply benchmark/src/runner.rs]: done

File: xline/src/storage/index.rs (hunk 6 above)
Comment (bsbds) [reply xline/src/storage/index.rs]: done

File: xline/src/server/command.rs (hunk 5 above)
Comment (bsbds) [reply xline/src/server/command.rs]: done

File: benchmark/src/runner.rs (hunk 3 above)
Comment (bsbds) [reply benchmark/src/runner.rs]: done

**Review by Phoenix500526 — APPROVED** [review]

**Review by iGxnon — APPROVED** [review]

**Pull request merged**